task/sast-snyk-check/0.3/sast-snyk-check.yaml

```diff
@@ -0,0 +1,233 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: sast-snyk-check
+spec:
+  description: >-
+    Scans source code for security vulnerabilities, including common issues such as SQL injection, cross-site scripting (XSS), and code injection attacks using Snyk Code, a Static Application Security Testing (SAST) tool.
+  results:
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+  params:
+    - name: SNYK_SECRET
+      description: Name of secret which contains Snyk token.
+      default: snyk-secret
+    - name: ARGS
+      type: string
+      description: Append arguments.
+      default: ""
+    - description: Image URL.
+      name: image-url
+      type: string
+      # In a future 0.4 version of the task, drop the default to make this required
```

Comment:
[The code](https://github.com/konflux-ci/build-definitions/blob/e95aec6a94bf6dcf532cd75de9a6b8b4e7f79fb1/task/sast-snyk-check/0.2/sast-snyk-check.yaml#L113) to skip the upload when `image-url` is empty is already being removed in 0.3 though.

Reply:
I am not sure If I understand your comment. I added the checks again as I don't know when the Konflux team is going to request `image-url` and `image-digest` to be mandatory. Does that make sense?

Reply:
I was trying to say that the (now reintroduced) check is needed to make the task succeed without the `image-url` parameter set.  The check makes the `upload` step exit with 0 in this case.  Without the check, the step would invoke `select-oci-auth` and `oras attach` with an empty string, which would fail and most likely cause the whole task to be marked as failed.  Looking at the code now, I am not sure why the task actually takes the `image-digest` parameter, which is not used for anything but the check.

Reply:
The last use of the `image-digest` value was removed in d1aa94f80c7b44d37c836ffe63650d15add2d60d.

Reply:
So it really seems it was removed. I have removed the references and have left the check for `IMAGE_URL`